### DISCUSSION TITLE
Application - Research Assistant

### DISCUSSION THREAD

**Discussion by mdlmarkham** [FlowiseAI/Flowise#3176] [Ideas]
I would like to build the back-end of a system like [Perplexity](https://www.perplexity.ai/) or [STORM](https://storm.genie.stanford.edu/) Ideally the tool would support research, encourage human review and re-direction, iterate, and finally generate a research product (paper, article, report etc.) that would then be entered into a vector database / knowledge base to be available for easy reference in the future.

**Comment by bonjohen** [comment]
I am doing a bit of research on that right now.
This link came up in my search. 

I am using this as a chance to organize my thoughts - maybe it will help someone.

I want it to fit entirely on a laptop - local LLMs and only use web for internet data.

 I have these pieces and planntontie them all together. 
* querying GoogleTrends. 
* I have identified 23 free journal article sites, fact checking sites, news sites, etc that have search built into their URL.
* the Metadata about search syntax and expected results in a JSON.
* A prompt that generates a list kf formatted URLs with provided query strings.
* A web query to get dom for a link from a search result.
* A process that constructs MDs from the dom.

I am doing this on a local Llama3.2 3b model.. (very snappy on a RTX 4070 desktop - acceptible on a RTX 1060) .
 It seems to be perfectly good at this sort of thing, and I think it should be a good "free" way to create research documents once I get a bit farther.
